### ISSUE TITLE
Error on kbox update. Error inspecting container

### ISSUE THREAD

**Issue by pi-ron** [kalabox/kalabox#749]
I've just updated code to 0.10.6 and ran `kbox update`, got this error message which references an old app I haven't used since way back when.

I tried `kbox cleanup` but that prompts for the update again.  I'll try dig around to find that boot 2 docker guide for inspecting and removing containers and see if it's lying around in there.

``````
info: VError: Error dispatching event listeners [post-stop]: Error inspecting container: kb_brazzen_data.: Error querying docker for list of containers.: Error initializing docker.: Error getting docker config.: Error running command "ssh,-o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i "/Users/RoamingMac/.ssh/boot2docker.kalabox.id_rsa" docker@localhost,sudo ifconfig eth1 10.13.37.42 netmask 255.255.255.0 broadcast 10.13.37.255 up".: code: 255, msg:
info: -- INSTALL FAILURE. --
``````

**Comment by pirog** [comment]
try running kbox update again with verbose on and see if it completes the second time through

**Comment by pi-ron** [comment]
Thanks Mike, I actually removed the trouble container, then re-ran (prior to seeing your response :-), got another trouble app error.  Removed that, then reran and update succeeded.

**Comment by pirog** [comment]
awesome! so you are good to go?

**Comment by pi-ron** [comment]
Back in action :-)

**Comment by pirog** [comment]
awesome! 

that said, this seems to be one of the lingering major issues in 0.10.6 based on error reporting so going to keep this open and add to this sprint.

**Comment by labboy0276** [comment]
If this helps any.  I got this again while creating huddle multidev (named zendesk) if you want to test it:

```
john@kalamuna:~/kalabox/apps/huddle$ kbox start
info: DOCKER =>  Creating ad hoc container "kalabox/debian:stable". name=kalabox_temp-5b854293-3999-4437-b515-8e021e52c723, Cmd=[bash], Tty=true, Image=kalabox/debian:stable, Env=[APPDOMAIN=huddle.kbox, KBOX_CODEDIR=code, KALABOX=true, APPNAME=huddle, SSH_KEY=pantheon.kalabox.id_rsa, GITUSER=John Ouellet, GITEMAIL=john@kalamuna.com, PANTHEON_ACCOUNT=john@kalamuna.com], VolumesFrom=[kb_huddle_data]
info: DOCKER =>  Ad hoc container created. 04713d18a0754523d30d490855e6d868b0441d0afd842e7b5e66c8bf58f0e475
info: DOCKER =>  Starting ad hoc container "04713d18a0754523d30d490855e6d868b0441d0afd842e7b5e66c8bf58f0e475". Binds=[/Users/john/kalabox/apps/huddle:/src:rw]
info: DOCKER =>  Ad hoc container started. 04713d18a0754523d30d490855e6d868b0441d0afd842e7b5e66c8bf58f0e475
info: DOCKER =>  Removing ad hoc container. 04713d18a0754523d30d490855e6d868b0441d0afd842e7b5e66c8bf58f0e475
info: DOCKER =>  Starting container 1cc0fb4a24dd3041476d7f1412a3d5061f9f21838669a548177506b0b62eaa1b. PublishAllPorts=true, Binds=[/Users/john/kalabox/apps/huddle:/src:rw]
info: DOCKER =>  Container started. 1cc0fb4a24dd3041476d7f1412a3d5061f9f21838669a548177506b0b62eaa1b
info: DOCKER =>  Starting container 914c6385c07b7d363cbe6f594c7e8d196d61f22c20c08a66cb70e836d5d2d14e. PublishAllPorts=true, Binds=[/Users/john/kalabox/apps/huddle:/src:rw]
info: DOCKER =>  Container started. 914c6385c07b7d363cbe6f594c7e8d196d61f22c20c08a66cb70e836d5d2d14e
info: DOCKER =>  Starting container 632945addc1dd51a5a1baca9fcde4c0de7da217e7d54d917d5ecb3bf212c0523. PublishAllPorts=true, Binds=[/Users/john/kalabox/apps/huddle:/src:rw]
info: DOCKER =>  Container started. 632945addc1dd51a5a1baca9fcde4c0de7da217e7d54d917d5ecb3bf212c0523
info: DOCKER =>  Starting container fc35cbadf3418445fcdabd082023237cc273d77773d42070c105c26c425d514b. PublishAllPorts=true, Binds=[/Users/john/kalabox/apps/huddle:/src:rw]
info: DOCKER =>  Container started. fc35cbadf3418445fcdabd082023237cc273d77773d42070c105c26c425d514b
info: DOCKER =>  Starting container afa3f4589f875918f3a9f42101f42176b6ccbb269a61955e72a01adb7d49300b. PublishAllPorts=true, Binds=[/Users/john/kalabox/apps/huddle:/src:rw]
info: DOCKER =>  Container started. afa3f4589f875918f3a9f42101f42176b6ccbb269a61955e72a01adb7d49300b
Error dispatching event listeners [post-start]: Error inspecting container: kb_huddle_redis.: Error initializing docker.: Error getting docker config.: Error running command "ssh,-o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i "/home/john/.ssh/boot2docker.kalabox.id_rsa" docker@localhost,sudo ifconfig eth1 10.13.37.42 netmask 255.255.255.0 broadcast 10.13.37.255 up".: code: 255, msg: 
VError: Error inspecting container: kb_huddle_redis.: Error initializing docker.: Error getting docker config.: Error running command "ssh,-o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i "/home/john/.ssh/boot2docker.kalabox.id_rsa" docker@localhost,sudo ifconfig eth1 10.13.37.42 netmask 255.255.255.0 broadcast 10.13.37.255 up".: code: 255, msg: 
    at /usr/local/bin/node_modules/kalabox-engine-docker/docker.js.jx:290:13
From previous event:
    at Object.inspect (/usr/local/bin/node_modules/kalabox-engine-docker/docker.js.jx:289:11)
From previous event:
    at /usr/local/bin/lib/engine.js.jx:422:29
From previous event:
    at Object.exports.inspect (/usr/local/bin/lib/engine.js.jx:421:4)
    at engine.list.map.concurrency (/usr/local/bin/lib/share.js.jx:33:19)
From previous event:
    at isAppRunning (/usr/local/bin/lib/share.js.jx:32:4)
    at /usr/local/bin/lib/share.js.jx:152:14
From previous event:
    at getDataContainers (/usr/local/bin/lib/share.js.jx:143:4)
    at getVolumes (/usr/local/bin/lib/share.js.jx:193:10)
    at Object.exports.restart (/usr/local/bin/lib/share.js.jx:416:17)
    at Object.fn (/usr/local/bin/plugins/kalabox-syncthing/events.js.jx:134:15)
    at /usr/local/bin/lib/core/events.js.jx:178:14
    at /usr/local/bin/lib/core/events.js.jx:177:24
From previous event:
    at /usr/local/bin/lib/core/events.js.jx:161:23
From previous event:
    at EventEngine._dispatch (/usr/local/bin/lib/core/events.js.jx:158:18)
    at /usr/local/bin/lib/core/events.js.jx:137:19
From previous event:
    at EventEngine.emit (/usr/local/bin/lib/core/events.js.jx:129:21)
From previous event:
    at Object.exports.emit (/usr/local/bin/lib/core/events.js.jx:261:4)
    at /usr/local/bin/lib/app.js.jx:759:24
From previous event:
    at Object.exports.start (/usr/local/bin/lib/app.js.jx:758:4)
    at Object.task.func (/usr/local/bin/plugins/kalabox-core/tasks/start.js.jx:24:18)
    at Object.exports.run (/usr/local/bin/lib/core/tasks.js.jx:876:15)
    at processTask (/usr/local/bin/bin/kbox.js.jx:170:16)
    at Object.<anonymous> (/usr/local/bin/bin/kbox.js.jx:233:12)
    at process._tickCallback (node.js:812:13)
From previous event:
    at main (/usr/local/bin/bin/kbox.js.jx:232:4)
    at Object.<anonymous> (/usr/local/bin/bin/kbox.js.jx:241:1)
    at Module._compile (module.js:518:26)
    at readX (module.js:1284:5)
    at Object.Module._extensions..jx (module.js:1711:10)
    at Module.load (module.js:345:36)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:377:17)
    at require (module.js:395:17)
    at Object.<anonymous> (_jx_source:17:3)
    at Module._compile (module.js:518:26)
    at Module.load (module.js:353:12)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:733:12)
    at startup (node.js:371:18)
    at node.js:1464:3
info: BOOT2DOCKER =>  Starting up. max=3
info: BOOT2DOCKER =>  Initializing boot2docker [1].
info: BOOT2DOCKER =>  Starting up. max=3
info: BOOT2DOCKER =>  Initializing boot2docker [1].
john@kalamuna:~/kalabox/apps/huddle$
```

**Comment by jdleonard** [comment]
I encountered something similar today, but not in the context of `kbox update`. Requests to Drupal were timing out so I...
```
JDs-MacBook-Pro:code jd$ kbox restart
info: BOOT2DOCKER =>  Starting up. maxRetries=3, max=3
info: BOOT2DOCKER =>  Initializing boot2docker [1].
info: BOOT2DOCKER =>  Bringing boot2docker up [1].
info: BOOT2DOCKER =>  Boot2docker is up.
Kalabox engine has been activated.
Error querying isRunning: "'2a20790b8e9a844396c257dc28c01c1b979b083ff732d77d9eb9f2776b354e7d'".: Error inspecting container: 2a20790b8e9a844396c257dc28c01c1b979b083ff732d77d9eb9f2776b354e7d.: Error querying docker for list of containers.: Failed after 5 retries. {"max":5,"backoff":500}: connect Unknown system errno 64
VError: Error inspecting container: 2a20790b8e9a844396c257dc28c01c1b979b083ff732d77d9eb9f2776b354e7d.: Error querying docker for list of containers.: Failed after 5 retries. {"max":5,"backoff":500}: connect Unknown system errno 64
    at /usr/local/bin/node_modules/kalabox-engine-docker/docker.js.jx:290:13
From previous event:
    at inspect (/usr/local/bin/node_modules/kalabox-engine-docker/docker.js.jx:289:11)
    at /usr/local/bin/node_modules/kalabox-engine-docker/docker.js.jx:306:14
From previous event:
    at Object.isRunning (/usr/local/bin/node_modules/kalabox-engine-docker/docker.js.jx:305:6)
From previous event:
    at /usr/local/bin/lib/engine.js.jx:349:29
From previous event:
    at Object.exports.isRunning.exports.isContainerRunning (/usr/local/bin/lib/engine.js.jx:348:4)
    at /usr/local/bin/node_modules/kalabox-services-kalabox/kalabox.js.jx:67:26
From previous event:
    at isServiceRunning (/usr/local/bin/node_modules/kalabox-services-kalabox/kalabox.js.jx:66:6)
    at serviceInfo.then.filter.concurrency (/usr/local/bin/node_modules/kalabox-services-kalabox/kalabox.js.jx:386:14)
From previous event:
    at Object.verify (/usr/local/bin/node_modules/kalabox-services-kalabox/kalabox.js.jx:385:6)
From previous event:
    at Object.exports.verify (/usr/local/bin/lib/services.js.jx:89:29)
    at exports.stop (/usr/local/bin/lib/app.js.jx:806:19)
    at Object.exports.restart (/usr/local/bin/lib/app.js.jx:859:10)
    at Object.task.func (/usr/local/bin/plugins/kalabox-core/tasks/restart.js.jx:15:18)
    at Object.exports.run (/usr/local/bin/lib/core/tasks.js.jx:876:15)
    at processTask (/usr/local/bin/bin/kbox.js.jx:170:16)
    at Object.<anonymous> (/usr/local/bin/bin/kbox.js.jx:233:12)
    at process._tickCallback (node.js:803:1)
From previous event:
    at main (/usr/local/bin/bin/kbox.js.jx:232:4)
    at Object.<anonymous> (/usr/local/bin/bin/kbox.js.jx:241:1)
    at Module._compile (module.js:595:24)
    at readX (module.js:1488:3)
    at Object.Module._extensions..jx (module.js:1921:8)
    at Module.load (module.js:416:30)
    at Function.Module._load (module.js:380:8)
    at Module.require (module.js:451:15)
    at require (module.js:469:13)
    at Object.<anonymous> (_jx_source:17:1)
    at Module._compile (module.js:595:24)
    at Module.load (module.js:425:6)
    at Function.Module._load (module.js:380:8)
    at Function.Module.runMain (module.js:813:8)
    at startup (node.js:374:8)
    at node.js:1461:1
```

After another `kbox stop` and `kbox start` (or two), I was able to get the app running again, but browser requests were resulting in "Error 503 Backend fetch failed".

So I `kbox rebuild` and `kbox start`, but still getting 503s.

**Comment by jdleonard** [comment]
Is there a way to get at the database, which has data that hasn't been backed up?

**Comment by labboy0276** [comment]
Hello

Well you can use normal mysql commands like:

mysql -u pantheon -h YOURINSTANCE.kbox -P 32799 pantheon

If the app is started, you can get the port by running kbox containers and
put it after the -P

On Tue, Jan 5, 2016 at 9:12 AM, JD Leonard <notifications@github.com> wrote:

> Is there a way to get at the database, which has data that hasn't been
> backed up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/749#issuecomment-169011325>.
>



-- 
*John Ouellet*
*Sr Developer & Support Manager*
415-470-3076
www.kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by jdleonard** [comment]
Thanks John. I knew there was a way, but I couldn't find it in the wiki.

Unfortunately I'm seeing zero tables when connecting (successfully) to the database, which is extra strange.

**Comment by labboy0276** [comment]
Hmmm

Try doing a kbox rebuild && kbox start and see if that helps any.

On Tue, Jan 5, 2016 at 11:42 AM, JD Leonard <notifications@github.com>
wrote:

> Thanks John. I knew there was a way, but I couldn't find it in the wiki.
>
> Unfortunately I'm seeing zero tables when connecting (successfully) to the
> database, which is extra strange.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/749#issuecomment-169056319>.
>



-- 
*John Ouellet*
*Sr Developer & Support Manager*
415-470-3076
www.kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by jdleonard** [comment]
No luck.

**Comment by labboy0276** [comment]
Can you try a kbox pull ?  grab the db or do you not want to annihilate
your local?

On Tue, Jan 5, 2016 at 1:05 PM, JD Leonard <notifications@github.com> wrote:

> No luck.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/749#issuecomment-169084087>.
>



-- 
*John Ouellet*
*Sr Developer & Support Manager*
415-470-3076
www.kalamuna.com

Stay connected with Kalamuna on: Twitter
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Ftwitter.com%2Fkalamuna&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>
, Linkedin
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2584606&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>,
and Facebook
<http://t.sidekickopen01.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1pwYhxW8qlTmx64kckdW8pTc_456dT6hf3BRzQ602?t=http%3A%2F%2Fwww.facebook.com%2Fkalamunatech&si=5428072445640704&pi=f16b515d-1f50-46c7-aebf-11a0c79916b9>

**Comment by jdleonard** [comment]
Was hoping to gain access to the contents of the database, but that's looking unlikely so I'm proceeding to annihilate. Thankfully not a ton of data lost.

**Comment by labboy0276** [comment]
@pirog I was helping one of our clients try and win Kalabox and they got this error

```
VError: Error running command "ssh,-o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i "/Users/lstrano/.ssh/boot2docker.kalabox.id_rsa" docker@localhost,sudo ifconfig eth1 10.13.37.42 netmask 255.255.255.0 broadcast 10.13.37.255 up".: code: 255, msg:
Error running step: {"name":"engine-docker-update-iso","description":"Updating B2D Iso","deps":[],"subscribes":["engine-up"],"first":false,"last":false,"isStep":true}: Error while shutting down.: Error dispatching event listeners [pre-down]: Error running command "ssh,-o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -p 2022 -i "/Users/lstrano/.ssh/boot2docker.kalabox.id_rsa" docker@localhost,sudo ifconfig eth1 10.13.37.42 netmask 255.255.255.0 broadcast 10.13.37.255 up".: code: 255, msg: 
```

Which seems similar to regular issue.  I told them to wait until 0.11.x anyway.

**Comment by pirog** [comment]
This should be resolved in the 0.11.0 release on Friday
